ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiConfigManagementService.scala

```diff
@@ -60,19 +53,27 @@ private[apiserver] final class ApiConfigManagementService private (
     logger.info("Getting time model")
     index
       .lookupConfiguration()
-      .map(_.fold(defaultConfigResponse) { case (_, conf) => configToResponse(conf) })
+      .flatMap {
+        case Some((_, configuration)) =>
+          Future.successful(configurationToResponse(configuration))
+        case None =>
+          logger.warn(
+            "Could not get the current time model. The index does not yet have the ledger configuration."
+          )
+          Future.failed(ErrorFactories.missingLedgerConfigOnConfigRequest())
+      }
```

Comment:
We need it to use `Future.failed` here. Can't do that with a `.map(_.fold(...))`.

Reply:
Why not use a throw?

Reply:
Because I expect `.map` to take a successful future and return a successful future, or take a failed future and return a failed future. I appreciate that `Future` will catch exceptions within its chained method calls and convert them to failed futures, but IMO this is specifically a backup plan and I don't like to rely on it.

Long story short, I don't like mixing my Scala monads and `throw`. I'd rather live in the Scala universe or the Java one, but not both within the same method.

We may disagree here. Happy to debate further.

Reply:
We do disagree but I don't have any hard opinions about this so feel free to leave it as is. 

As for my reasoning:

I would expect that we would use `flatMap` when there's an actual Future context, as this would also make me do a double take when reading the code to make sure I'm aware of what `complications` we're merging, and in this case it's an artificial usage. 

I do agree that throw is not the best but imo we should return a `Future[Either[L, R]]` so that we don't mix exceptions with error handling (and fold the future before returning it), or do accept that scala future just ain't pretty.

Reply:
Yeah, I think you're right. Bigger change, but perhaps one we need to consider in the long run. I find it really tough to figure out all the places these API service implementations can fail.